app/controllers/facility_user_reservations_controller.rb

```diff
@@ -0,0 +1,52 @@
+class FacilityUserReservationsController < ApplicationController
+
+  admin_tab :all
+  before_action :init_current_facility
+  before_action :authenticate_user!
+  before_action :check_acting_as
+  before_action :load_user
+
+  load_and_authorize_resource class: Reservation
+
+  layout "two_column"
+
+  # GET /facilities/:facility_id/users/:user_id/reservations
+  def index
+    @order_details = user_order_details
+                     .where("orders.ordered_at IS NOT NULL")
+                     .order("orders.ordered_at DESC")
+                     .paginate(page: params[:page])
+  end
+
+  # PUT /facilities/:facility_id/users/:user_id/reservations/:reservation_id/cancel
+  def cancel
+    order_detail = user_order_details.find(params[:reservation_id])
```

Comment:
I hope it isn't too confusing to call this `reservation_id` when it's really an `order_detail_id`.